src/helpers/productsSorter.ts

```diff
@@ -0,0 +1,32 @@
+import { IProduct } from 'models/IProduct';
+import { byTitle, byPrice, byRating, ascendOrder } from '@constants';
+import { compareTitleAscend, compareTitleDescend } from './compareFunctions';
+import { comparePriceAscend, comparePriceDescend } from './compareFunctions';
+import { compareRatingAscend, compareRatingDescend } from './compareFunctions';
+
+const productsSorter = (products: IProduct[], sortOption: string[]) => {
+  const [sortBy, order] = sortOption;
+  let newProducts = [...products];
+
+  switch (sortBy) {
+    case byTitle:
+      order === ascendOrder
+        ? (newProducts = newProducts.sort(compareTitleAscend))
+        : (newProducts = newProducts.sort(compareTitleDescend));

```

Comment:
3 times almost the same condition. someFn = (ascend, descend) =>  order === ascendOrder
        ? (newProducts = newProducts.sort(ascend))
        : (newProducts = newProducts.sort(descend));